pkg/controller/master/backup/restore.go

```diff
@@ -220,11 +273,117 @@ func (h *RestoreHandler) PersistentVolumeClaimOnChange(key string, pvc *corev1.P
 		return nil, nil
 	}
 
+	volumeName, err := getVolumeName(pvc)
+	if err != nil {
+		return nil, err
+	}
+
+	volume, err := h.volumeCache.Get(util.LonghornSystemNamespaceName, volumeName)
+	if err != nil {
+		return nil, nil
+	}
+
+	volumeCopy := volume.DeepCopy()
+	if volumeCopy.Annotations == nil {
+		volumeCopy.Annotations = make(map[string]string)
+	}
+
+	volumeCopy.Annotations[pvcNameSpaceAnnotation] = pvc.Namespace
+	volumeCopy.Annotations[pvcNameAnnotation] = pvc.Name
+	volumeCopy.Annotations[restoreNameAnnotation] = restoreName
+
+	if !reflect.DeepEqual(volume, volumeCopy) {
+		if _, err := h.volumes.Update(volumeCopy); err != nil {
+			return nil, err
+		}
+	}
+
 	logrus.Debugf("handling PVC updating %s/%s", pvc.Namespace, pvc.Name)
 	h.restoreController.EnqueueAfter(pvc.Namespace, restoreName, 5*time.Second)
 	return nil, nil
 }
 
+func (h *RestoreHandler) resolveLHEngineRef(lhEngine *lhv1beta2.Engine) (*lhv1beta2.Volume, error) {
+	refs := lhEngine.GetOwnerReferences()
+	if len(refs) == 0 {
+		return nil, fmt.Errorf("engine %s missing volume owner", lhEngine.Name)
+	}
+
+	ref := refs[0]
```

Comment:
Can we just get volume name from `lhEngine.Spec.VolumeName`?

Reply:
Updated per suggestion, thanks.